components/request_otr/browser/request_otr_rule.h

```diff
@@ -0,0 +1,54 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_REQUEST_OTR_BROWSER_REQUEST_OTR_RULE_H_
+#define BRAVE_COMPONENTS_REQUEST_OTR_BROWSER_REQUEST_OTR_RULE_H_
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/containers/flat_set.h"
+#include "base/json/json_value_converter.h"
+#include "base/types/expected.h"
+#include "base/values.h"
+#include "components/prefs/pref_service.h"
+#include "extensions/common/url_pattern_set.h"
+
+class GURL;
+
+namespace request_otr {
+
+class RequestOTRRule {
+ public:
+  RequestOTRRule();
+  ~RequestOTRRule();
+
+  // Registers the mapping between JSON field names and the members in this
+  // class.
+  static void RegisterJSONConverter(
```

Comment:
the preferred way to convert json is actually `tools/json_schema_compiler/json_schema_api.gni` - there are examples in brave_wallet

Reply:
When did this change? I see JSONValueConverter in debounce_rule, ad_block_subscription_service_manager, filter_list_catalog_entry.cc

Reply:
apparently it has been the case for a long time in upstream, we just weren't aware of it. We can leave this, but just fyi for next time.